portality/forms/application_processors.py

```diff
@@ -300,7 +301,8 @@ def finalise(self, account, save_target=True, email_alert=True):
                 owner.save()
 
                 # for all acceptances, send an email to the owner of the journal
-                self._send_application_approved_email(j.bibjson().title, owner.name, owner.email, self.source.current_journal is not None)
+                if has_request_context():
+                    self._send_application_approved_email(j.bibjson().title, owner.name, owner.email, self.source.current_journal is not None)
```

Comment:
I have a more concrete objection to this one - we're defining the behaviour of the code according to its request context, even though the request context has no particular bearing on the execution of the code it protects.  If the code would fail in the absence of a request context, fair enough (notwithstanding comment above about our abstraction here).  But this code, I think, would succeed, so what we really are saying is that there are circumstances where during acceptance we don't want to send the approved email, and in which case that should be a switch that's passed in to the function, not assumed based on call context.

Reply:
Agreed, I was working around the fact that we'd copied across from formcontext here, so wasn't going into signatures. However it appears there's an `email_alert=True` flag on the function already which I'd failed to notice. I might use that, depending on prior use.

Reply:
Yes, that's better, if we can use that flag then this is good to go